core/trino-main/src/main/java/io/trino/connector/CatalogConnector.java

```diff
@@ -89,6 +105,8 @@ public ConnectorServices getMaterializedConnector(CatalogHandleType type)
 
     public void shutdown()
     {
+        long bytes = catalogConnector.getConnector().getExtraMemoryRequirement();
```

Comment:
This is problematic. There's no guarantee that `getExtraMemoryRequirements()` returns the same value across multiple calls, which could result in memory leaks or freeing more than was allocated.

Reply:
Why not? This is a method I created and the variable is defined only once - in the creation of the author

Reply:
The `getExtraMemoryRequirements()` method is implemented by connectors. How do you guarantee that arbitrary connectors return the same value across all invocations?

Reply:
The implementation is by connector but the invocation is by Catalog and different catalog can return different value base on its configuration